Formula/gomigrate.rb

```diff
@@ -0,0 +1,26 @@
+class Gomigrate < Formula
+  desc "Database migrations CLI tool"
+  homepage "https://github.com/golang-migrate/migrate"
+  url "https://github.com/golang-migrate/migrate/archive/v3.3.0.tar.gz"
+  sha256 "59ade052631d6a064d3ac71df5f03c120c3029210d8d13972b6c5db7558c490f"
+
+  depends_on "dep" => :build
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/golang-migrate/migrate").install buildpath.children
+
+    # Build and install cli as "migrate"
+    cd "src/github.com/golang-migrate/migrate" do
+      system "dep", "ensure", "-vendor-only"
+      system "make", "build-cli", "VERSION=v#{version}"
+      bin.install "cli/build/migrate.darwin-amd64" => "migrate"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    assert_match "v3.3.0", shell_output("#{bin}/migrate -version 2>&1")
```

Comment:
The test needs to exercise and verify the functionality of the software.

Reply:
functional test requires external service, would you suggest a way to do that?

Reply:
You can add `depends_on "foo" => :test` if necessary.

Reply:
(for instance see the php formula)